### PR TITLE
Vingette for step by step guide how to perform t-test.

### PR DESCRIPTION
**Pull request by Orbmac**
# Title: Write a very short summary here
**Problem:** Write a description of your problem here.

**Solution:** Write a short summary of your solution.

**Key Features:**

* Key feature 1
* Key feature 2
* ...

## Checklist
- [x] Make sure you are requesting to **pull a feature/bugfix branch** (right side). This should not be main or develop.
- [x] Make sure you are make a pull request against either **main or develop** (left side). (Requesting to main should be reserved for bugfixes and new releases)
- [ ] Add or update unit tests (if applicable)
- [ ] Check your code with any unit tests (Run devtools::check() locally)
- [ ] Add neccessary documentation (if applicable)

## Type of changes

What type of changes does your code introduce?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue) (link the issue on the right)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Code style update (formatting, renaming)
- [X] Documentation Update 
- [ ] Other (explain)

## Further comments
If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, any questions you have, etc...

Consider linking any issues (#issue-number ) or adding @mentions to ensure specific people see it.

### CODE REVIEW

**Comment by kathy-nevola** [comment]
Maybe add paired t-test vs unpaired t-test or will that be a separate vignette?

**Comment by kathy-nevola** [comment]
@Orbmac made some edits to clarify the wording, let me know if this makes sense to you. Im struggling to figure out how references work in vignettes and I cant make figure captions work, so if you can help with those issues that would be great,

**Comment by kathy-nevola** [comment]
Suggestion from user regarding clarity using function: specific pair_id column format and the fact that each group needed at least 16 samples

File: OlinkAnalyze/vignettes/study_design_2groups.Rmd [hunk 1]
```diff
@@ -0,0 +1,359 @@
...
+  filter(!str_detect(SampleID, 'CONTROL'))
+df_QC <- OlinkAnalyze::npx_data1 %>% 
+  filter(!str_detect(SampleID, 'CONTROL'))
+
+
+```
+
+# Introduction  
+
+This vignette is designed to assist users working with two distinct groups using t-test. This guide aims to provide you with a comprehensive overview of the steps involved in analyzing and comparing data from two groups in Olink data using the Olink R package. These two groups can encompass a variety of categories, such as healthy and diseased individuals, two different types of a condition, or male and female subjects. The objective of the analysis is to identify the differing proteins between the two groups.
```
Comment (kathy-nevola) [inline OlinkAnalyze/vignettes/study_design_2groups.Rmd:36]:
```suggestion
This vignette is designed to assist users working with two distinct groups using t-test. This guide aims to provide you with a comprehensive overview of the steps involved in analyzing and comparing data from two groups in Olink data using Olink's R package, Olink Analyze. These two groups can encompass a variety of categories, such as healthy and diseased individuals, two different types of a condition, or male and female subjects. The objective of the analysis is to identify the differing proteins between the two groups.
```

File: OlinkAnalyze/vignettes/study_design_2groups.Rmd [hunk 2]
```diff
@@ -0,0 +1,359 @@
...
+
+By following this guide, you will gain a solid understanding of how to effectively compare and contrast two groups, detect differences, and ultimately derive valuable conclusions from your data. Example code will be shown in this vignette, however it is still recommended to read the documentation for each function as not all options will be covered here. This can be done with the _help_ command.
+
+```{r, eval=FALSE}
+help("functionName")
+```
+
+If you have more that two groups in your data, or a longitudinal study see our other study design vignettes.
```
Comment (kathy-nevola) [inline OlinkAnalyze/vignettes/study_design_2groups.Rmd:44]:
```suggestion
If you have more that two groups in your data, or a longitudinal study, see our other study design vignettes. 
```

File: OlinkAnalyze/vignettes/study_design_2groups.Rmd [hunk 3]
```diff
@@ -0,0 +1,359 @@
...
+After installing the package, the package needs to be loaded. Several packages from the tidyverse suite of packages are also used throughout this vignette, such as dplyr and stringr. Each individual package can be loaded using the _library_ function (`library("package")`) or the entire tidyverse suite of packages can be loaded as shown below.
+
+```{r, eval=FALSE}
+library(OlinkAnalyze)
+library(tidyverse)
+```
+
+# Study overview
+
+This vignette will help you if you have Olink data and are unsure on how to compare your two groups. The Olink data file shouldn't be tempered with before loading it into R, as that may cause the data to not be recognized and loaded by the `read_NPX` function. Using the read_NPX function will always give you your data in a long format, where each row represent one sample and one protein. To load in your data in R use the following code:
```
Comment (kathy-nevola) [inline OlinkAnalyze/vignettes/study_design_2groups.Rmd:63]:
```suggestion
This vignette will help you if you have Olink data and are unsure on how to compare your two groups. The Olink data file shouldn't be tempered with before loading it into R, as that may cause the data to not be recognized and loaded by the `read_NPX` function. Using the read_NPX function will always give you your data in a long format, where each row represent one sample and one protein. To load in your data in R, use the following code:
```

File: OlinkAnalyze/vignettes/study_design_2groups.Rmd [hunk 4]
```diff
@@ -0,0 +1,359 @@
...
+
+```{r message=FALSE, eval=FALSE}
+data <- read_NPX("~/NPX_file_location.csv")
+# Argument should be the location of your NPX data. Data are usually in parquet, csv, xlsx or zip format depending on the product.
```
Comment (kathy-nevola) [inline OlinkAnalyze/vignettes/study_design_2groups.Rmd:67]:
```suggestion
# Argument should be the location of your NPX data. 
# Data are usually in parquet, csv, xlsx or zip format. 
# The file format will depend on the product.
```

File: OlinkAnalyze/vignettes/study_design_3groups.Rmd [hunk 5]
```diff
@@ -0,0 +1,351 @@
...
+df <- OlinkAnalyze::npx_data1 %>% 
+  filter(!str_detect(SampleID, 'CONTROL'))
+df_QC <- OlinkAnalyze::npx_data1 %>% 
+  filter(!str_detect(SampleID, 'CONTROL'))
+
+```
+
+# Introduction  
+
+This vignette is designed to assist users working with 3 or more distinct groups using ANOVA. This guide aims to provide you with a comprehensive overview of the steps involved in analyzing and comparing data from multiple groups in Olink data using the Olink R package.  These groups can encompass a variety of categories, such as disease subtypes, treatment regimens,or genetic variations. The objective of the analysis is to identify for which proteins the groups differ from at least one another.
```
Comment (kathy-nevola) [inline OlinkAnalyze/vignettes/study_design_3groups.Rmd:36]:
```suggestion
This vignette is designed to assist users working with 3 or more distinct groups using ANOVA. This guide aims to provide you with a comprehensive overview of the steps involved in analyzing and comparing data from multiple groups in Olink data using Olink's R package, Olink Analyze.  These groups can encompass a variety of categories, such as disease subtypes, treatment regimens, or genetic variations. The objective of the analysis is to identify proteins whose expressions differ between at least 2 of the groups. 
```

File: OlinkAnalyze/vignettes/study_design_3groups.Rmd [hunk 6]
```diff
@@ -0,0 +1,351 @@
...
+library(tidyverse)
+```
+
+# Study overview
+
+This vignette will help you if you have Olink data and are unsure on how to compare your groups. The Olink data file shouldn't be tempered with before loading it into R, as that may cause the data to not be recognized and loaded by the `read_NPX` function. Using the read_NPX function will always give you your data in a long format, where each row represent one sample and one protein. To load in your data in R use the following code:
+
+```{r message=FALSE, eval=FALSE}
+data <- read_NPX("~/NPX_file_location.csv")
+# Argument should be the location of your NPX data. Data are usually in parquet, csv, xlsx or zip format depending on the product.
```
Comment (kathy-nevola) [inline OlinkAnalyze/vignettes/study_design_3groups.Rmd:67]:
```suggestion
# Argument should be the location of your NPX data. 
# Data are usually in parquet, csv, xlsx or zip format.
# The data format depends on the product.
```

File: OlinkAnalyze/vignettes/study_design_3groups.Rmd [hunk 7]
```diff
@@ -0,0 +1,351 @@
...
+               longtable = T,
+               row.names = F,
+               caption = "**Table 4.** ANOVA post hoc results of the 5 most significant contrasts and proteins. Some coloumns have been left out due to space.",
+               label = "manifest")
+```  
+
+
+## Plots
+
+To illustrate the results from a ANOVA boxplots are a very effective way. These visualizations provide valuable insights into the differences between the groups groups and help in understanding the significance and effect sizes of the observed effects.
```
Comment (kathy-nevola) [inline OlinkAnalyze/vignettes/study_design_3groups.Rmd:244]:
```suggestion
Boxplots are a very effective way to illustrate the results from an ANOVA. These visualizations provide valuable insights into the differences between the groups and help in understanding the significance and effect sizes of the observed effects.
```

File: OlinkAnalyze/vignettes/study_design_3groups.Rmd [hunk 8]
```diff
@@ -0,0 +1,351 @@
...
+
+To generate boxplots the `olink_boxplot` function can be used. The thick black horizontal line within each box indicates the median, the colored area - the interquartile range, the ends of the vertical middle line represent the minimum and the maximum values, and the black dots indicate potential assay outliers (beyond the minumum or maximum valuses). Boxplots of the top 6 significant proteins are plotted in **Figure 3**.
```
Comment (kathy-nevola) [inline OlinkAnalyze/vignettes/study_design_3groups.Rmd:246]:
```suggestion
To generate boxplots, the `olink_boxplot` function can be used. The thick black horizontal line within each box indicates the median, the colored area - the interquartile range, the ends of the vertical middle line represent the minimum and the maximum values, and the black dots indicate potential assay outliers (beyond the minumum or maximum valuses). Boxplots of the top 6 significant proteins are plotted in **Figure 3**. 
```

File: OlinkAnalyze/vignettes/study_design_3groups.Rmd [hunk 9]
```diff
@@ -0,0 +1,351 @@
...
+# Saving results
+
+Having the results saved in csv files (or other format) and the plots in pdf (or other format) can making the results easier to work with, especially if the result should be sent to other people without access to R. 
+
+The following code can be used to save the results as a csv file.
+
+```{r message=FALSE, eval=F}
+
+write.table(anova_results,
+            paste0('ANOVAResults.csv'),
```
Comment (kathy-nevola) [inline OlinkAnalyze/vignettes/study_design_3groups.Rmd:274]: do we need the paste0 here?

File: OlinkAnalyze/vignettes/study_design_3groups.Rmd [hunk 10]
```diff
@@ -0,0 +1,351 @@
...
+            sep=';',row.names = F, quote = F)
+
+write.table(anova_posthoc_results,
+            paste0('ANOVA_PostHoc_Results.csv'),
```
Comment (kathy-nevola) [inline OlinkAnalyze/vignettes/study_design_3groups.Rmd:278]: same here. is paste0 needed?

File: OlinkAnalyze/vignettes/study_design_mixedmodel.Rmd [hunk 11]
```diff
@@ -0,0 +1,287 @@
...
+df <- OlinkAnalyze::npx_data1 %>% 
+  filter(!str_detect(SampleID, 'CONTROL'))
+df_QC <- OlinkAnalyze::npx_data1 %>% 
+  filter(!str_detect(SampleID, 'CONTROL'))
+
+```
+
+# Introduction  
+
+This vignette is designed to assist users working with longitudinal samples divided across multiple groups using linear mixed models. This guide aims to provide you with a comprehensive overview of the steps involved in analyzing Olink data when the samples are measured over time in different groups using the Olink R package. An example can be samples measured over time where half the subjects have been revived a treatment and the other half have not.
```
Comment (kathy-nevola) [inline OlinkAnalyze/vignettes/study_design_mixedmodel.Rmd:36]:
```suggestion
This vignette is designed to assist users working with longitudinal samples divided across multiple groups using linear mixed models. This guide aims to provide you with a comprehensive overview of the steps involved in analyzing Olink data when the samples are measured over time in different groups using Olink's R package, Olink Analyze. An example can be samples measured over time where half the subjects have been revived a treatment and the other half have not. 
```

File: OlinkAnalyze/vignettes/study_design_mixedmodel.Rmd [hunk 12]
```diff
@@ -0,0 +1,287 @@
...
+library(tidyverse)
+```
+
+# Study overview
+
+This vignette will help you if you have Olink data and are unsure on how to compare your longitudinal samples spread across multiple groups.  The Olink data file shouldn't be tempered with before loading it into R, as that may cause the data to not be recognized and loaded by the `read_NPX` function. Using the read_NPX function will always give you your data in a long format, where each row represent one sample and one protein. To load in your data in R use the following code:
+
+```{r message=FALSE, eval=FALSE}
+data <- read_NPX("~/NPX_file_location.csv")
+# Argument should be the location of your NPX data. Data are usually in parquet, csv, xlsx or zip format depending on the product.
```
Comment (kathy-nevola) [inline OlinkAnalyze/vignettes/study_design_mixedmodel.Rmd:67]:
```suggestion
# Argument should be the location of your NPX data. 
# Data are usually in parquet, csv, xlsx or zip format.
# Data file format depends on the product.
```

File: OlinkAnalyze/vignettes/study_design_mixedmodel.Rmd [hunk 13]
```diff
@@ -0,0 +1,287 @@
...
+```{r message=FALSE, eval=T}
+df %>%  
+  select(SampleID, Treatment, Time, Subject) %>% 
+  unique() %>%
+  select(Treatment, Time) %>% 
+  table
+
+```  
+
+LMM assesses the null hypothesis that no significant effect of the independent variables on the dependent variable. If the p-value associated with the test is lower than a predetermined significance level (typically 0.05), we can reject the null hypothesis and infer a significant effect of at least one independent variable on the dependent variable. This indicates that the independent variables, included in the model, have a statistically significant impact on explaining the variation in the dependent variable.
```
Comment (kathy-nevola) [inline OlinkAnalyze/vignettes/study_design_mixedmodel.Rmd:182]:
```suggestion
LMM assesses the null hypothesis that there is no significant effect of the independent variables on the dependent variable. If the p-value associated with the test is lower than a predetermined significance level (typically 0.05), we can reject the null hypothesis and infer a significant effect of at least one independent variable on the dependent variable. This indicates that the independent variables, included in the model, have a statistically significant impact on explaining the variation in the dependent variable.
```

File: OlinkAnalyze/vignettes/study_design_mixedmodel.Rmd [hunk 14]
```diff
@@ -0,0 +1,287 @@
...
+Conversely, if the p-value exceeds the significance level, we do not have enough evidence to reject the null hypothesis, suggesting an absence of a statistically significant effect of the independent variables on the dependent variable in the context of the linear mixed model.
+
+The LMM also take into account the random effect, which is a critical component of the model. Random effects are used to model the variability within groups or clusters in the data, accounting for correlations and hierarchical structures that might exist. These random effects are essential in capturing the dependencies and correlations in the data that cannot be explained solely by the independent variables. The random effect are usually subject for Olink studies.
```
Comment (kathy-nevola) [inline OlinkAnalyze/vignettes/study_design_mixedmodel.Rmd:185]:
```suggestion
The LMM also take into account the random effect, which is a critical component of the model. Random effects are used to model the variability within groups or clusters in the data, accounting for correlations and hierarchical structures that might exist. These random effects are essential in capturing the dependencies and correlations in the data that cannot be explained solely by the independent variables. The random effect is usually within subject correlation for Olink studies. 
```

File: OlinkAnalyze/vignettes/study_design_mixedmodel.Rmd [hunk 15]
```diff
@@ -0,0 +1,287 @@
...
+lmer_posthoc_results_Time %>%
+  head(5)  %>% 
+  select(Assay, OlinkID, UniProt, Panel, contrast ,estimate, Adjusted_pval, Threshold) %>% 
+  knitr::kable(format = "html",
+               booktabs = T,
+               linesep = "",
+               digits = 4,
+               longtable = T,
+               row.names = F,
+               caption = "**Table 4.** ANOVA post hoc results of the 5 most significant contrasts and proteins. Some coloumns have been left out due to space.",
```
Comment (kathy-nevola) [inline OlinkAnalyze/vignettes/study_design_mixedmodel.Rmd:243]:
```suggestion
               caption = "**Table 4.** ANOVA post hoc results of the 5 most significant contrasts and proteins. Some columns have been left out due to space.",
```

File: OlinkAnalyze/vignettes/study_design_mixedmodel.Rmd [hunk 16]
```diff
@@ -0,0 +1,287 @@
...
+               label = "manifest")
+```  
+
+
+## Plots
+
+To illustrate the results from a LMM a point range plot are a common way. These visualizations provide valuable insights into the differences between the groups and help in understanding the significance and effect sizes of the observed effects.
```
Comment (kathy-nevola) [inline OlinkAnalyze/vignettes/study_design_mixedmodel.Rmd:250]:
```suggestion
A point range plot is a common way to illustrate the results from an LMM. These visualizations provide valuable insights into the differences between the groups and help in understanding the significance and effect sizes of the observed effects.
```

File: OlinkAnalyze/vignettes/study_design_mixedmodel.Rmd [hunk 17]
```diff
@@ -0,0 +1,287 @@
...
+# Saving results
+
+Having the results saved in csv files (or other format) and the plots in pdf (or other format) can making the results easier to work with, especially if the results are being shared with other people that do not have access to R. 
+
+The following code can be used to save the results as a csv file.
+
+```{r message=FALSE, eval=F}
+
+write.table(lmer_results,
+            paste0('LMER_Results.csv'),
```
Comment (kathy-nevola) [inline OlinkAnalyze/vignettes/study_design_mixedmodel.Rmd:280]: same question as before with paste0

File: OlinkAnalyze/vignettes/study_design_mixedmodel.Rmd [hunk 18]
```diff
@@ -0,0 +1,287 @@
...
+            sep=';',row.names = F, quote = F)
+
+write.table(lmer_posthoc_results_Time, # each term should be saved
+            paste0('LMER_posthoc_Time_Results.csv'),
+            sep=';',row.names = F, quote = F)
+
+```
```
Comment (kathy-nevola) [inline OlinkAnalyze/vignettes/study_design_mixedmodel.Rmd:287]: Add contact us section

File: OlinkAnalyze/vignettes/study_design_mixedmodel.Rmd [hunk 19]
```diff
@@ -0,0 +1,287 @@
...
+
+The following code can be used to save the results as a csv file.
+
+```{r message=FALSE, eval=F}
+
+write.table(lmer_results,
+            paste0('LMER_Results.csv'),
+            sep=';',row.names = F, quote = F)
+
+write.table(lmer_posthoc_results_Time, # each term should be saved
```
Comment (kathy-nevola) [inline OlinkAnalyze/vignettes/study_design_mixedmodel.Rmd:283]: same question with paste0

**Review by kathy-nevola — CHANGES_REQUESTED** [review]

File: OlinkAnalyze/vignettes/study_design_mixedmodel.Rmd [hunk 20]
```diff
@@ -0,0 +1,287 @@
...
+title: "A Guide to Comparative Analysis of logitudinal data across muliple groups"
+author: "Olink DS team"
+date: "`r Sys.Date()`"
+output: 
+  html_vignette:
+    latex_engine: xelatex
+    toc: true
+    toc_depth: 2
+vignette: >
+  %\VignetteIndexEntry{Olink® Analyze Vignette}
```
Comment (kathy-nevola) [inline OlinkAnalyze/vignettes/study_design_mixedmodel.Rmd:11]:
```suggestion
  %\VignetteIndexEntry{A Guide to Comparative Analysis of longitudinal data across multiple groups}
```

File: OlinkAnalyze/vignettes/study_design_mixedmodel.Rmd [hunk 21]
```diff
@@ -0,0 +1,287 @@
+---
+title: "A Guide to Comparative Analysis of logitudinal data across muliple groups"
```
Comment (kathy-nevola) [inline OlinkAnalyze/vignettes/study_design_mixedmodel.Rmd:2]:
```suggestion
title: "A Guide to Comparative Analysis of longitudinal data across multiple groups"
```

File: OlinkAnalyze/vignettes/study_design_3groups.Rmd [hunk 22]
```diff
@@ -0,0 +1,351 @@
...
+title: "A Guide to Comparative Analysis of Muliple Groups with Olink Data"
+author: "Olink DS team"
+date: "`r Sys.Date()`"
+output: 
+  html_vignette:
+    latex_engine: xelatex
+    toc: true
+    toc_depth: 2
+vignette: >
+  %\VignetteIndexEntry{A Guide to Comparative Analysis of Muliple Groups with Olink Data}
```
Comment (kathy-nevola) [inline OlinkAnalyze/vignettes/study_design_3groups.Rmd:11]:
```suggestion
  %\VignetteIndexEntry{A Guide to Comparative Analysis of Multiple Groups with Olink Data}
```

**Review by Orbmac — COMMENTED** [review]
All changed added

**Review by kathy-nevola — APPROVED** [review]
LGTM

**Pull request closed without merging**